homeassistant/components/downloader/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow for Downloader integration."""
+from __future__ import annotations
+
+import os
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import _LOGGER, CONF_DOWNLOAD_DIR, DEFAULT_NAME, DOMAIN
+
+
+class DownloaderConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Downloader."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            try:
+                await self._validate_input(user_input)
+            except DirectoryDoesNotExist:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(title=DEFAULT_NAME, data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_DOWNLOAD_DIR): cv.string,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by configuration file."""
+
+        return await self.async_step_user(user_input)
```

Comment:
We should not use the user step here since it doesn't abort. The import flow needs to abort on error since there won't be any forms to show errors to the user.

We also need to complete test coverage for when the import flow aborts and what repair issues are created.